doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,267 @@
+
+If the EQ_FIXED flag is false or not present, then the SI param generation flow will come out of TX_SETTING and continue with the RX_SETTING block. But if the EQ_FIXED flag is true for TX_SETTING, then we need to disable AdaptiveInputEqEnableTx.
+
+The TX Input EQ register control: Page 10h Byte 153 – 159 
+
+| **Byte**       | **Field Name**                            |
+| -------------- | ----------------------------------------- |
+| 153            | AdaptiveInputEqEnableTx1..8 (lane 1-8)    |
+| 154 - 155      | AdaptiveInputEqRecallTx1..8 (lane 1-8)    |
+| 156 - 159      | FixedInputEqTargetTx1..8    (lane 1-8)    |
+
+### 3.2 RX_SETTING:   
+The RX_SETTING block contains the same sections as TX_SETTING, but the EQ_FIXED flag should always be true. The SI settings can be directly written and applied for RX output equalization.
+
+The RX Output EQ register control: Page 10h Byte 162 – 173
+
+| **Byte**       | **Field Name**                            |
+| -------------- | ----------------------------------------- |
+| 162 - 165      | OutputEqPreCursorTargetRx1..8 (lane 1-8)  |
+| 166 - 169      | OutputEqPostCursorTargetRx1..8 (lane 1-8) |
+| 170 - 173      | OutputAmplitudeTargetRx1..8    (lane 1-8) |
+
+### 3.3 GLOBAL_MEDIA_SETTINGS:  
+This block's first level of identification will be the range of port numbers. The ports can be defined as a range of 0-31 or a list of multiple ports: 1, 2, 3, or a list of ports in the range of 5–10, 25–31, matching the index number in the port_config.ini file. This port range will have a unique defined lane speed, which will have unique vendor and vendor part number entries supporting this speed. Module key will be created based on speed and vendor details.
+
+Each vendor will have per-lane SI param attribute entries applicable for the identified port + speed for the platform vendor. This value will be searched through the module key.
+
+### 3.4 PORT_MEDIA_SETTINGS:  
+The entries in this block will be unique single port numbers. The control of SI attribute list generation search will reach the PORT_MEDIA_SETTINGS block only when no attribute list is generated in the GLOBAL_MEDIA_SETTINGS block.   
+
+There will be unique speed and vendor/vendor_PN entries in each identified port block.
+
+Default values can be platform defaults for multiple vendors in each section.
+
+### 3.5 List of standard TX RX SI parameters
+-  SI_PARAM_TX_INPUT_EQ 
+-  SI_PARAM_RX_OUTPUT_PRE 
+-  SI_PARAM_RX_OUTPUT_POST 
+-  SI_PARAM_RX_OUTPUT_AMP
+
+### 3.6 Sample Optics SI setting file:
+```
+{
+  “TX_SETTING”: { 
+      “EQ_FIXED”: “false”, 
+      “GLOBAL_MEDIA_SETTINGS” : {}, 
+      “PORT_MEDIA_SETTINGS” : {} 
+  },
+  
+  “RX_SETTING”: { 
+      “EQ_FIXED”: “true”, 
+      “GLOBAL_MEDIA_SETTINGS” : { 
+          “1-20”: { 
+              “100G_SPEED”: { 
+                  “CREDO  -CAC82X321M2MC0HW”: { 
```

Comment:
@AnoopKamath can you specify how did you arrive at this key? Is this vendor+PN? I see some space between vendor name and PN

Reply:
Yes. Its the Vendor Key = Vendor name + Vendor part number. Fixed spacing issue